src/Models/BaseModel.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Tighten\NovaStripe\Models;
+
+use Exception;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Carbon;
+use Tighten\NovaStripe\Services\StripeClientService;
+
+abstract class BaseModel extends Model
+{
+    protected $guarded = [];
+
+    public function sync(): array
+    {
+        $items = $this->getService()->all([
+            'limit' => 100,
+            'expand' => $this->expand ?? [],
+        ]);
+
+        $records = [];
+        foreach ($items->autoPagingIterator() as $item) {
+            $records[] = $this->prepareForInsert($item);
+        }
```

Comment:
I think it may be a teensy bit faster on huge data sets to keep it this way, so I think you should probably keep it, but just FYI a collection map would be a bit cleaner:

```suggestion
        $records = collect($items->autoPagingIterator())->map(function ($item) {
            return $this->prepareForInsert($item);
        })->toArray();
```

Reply:
Cleaner indeed! Updated 👍